ts/WoltLabSuite/Core/Acp/Ui/Package/Update/Manager.ts

```diff
@@ -0,0 +1,200 @@
+
+const enum ResponseType {
+  AuthorizationRequired = "authorizationRequired",
+  Conflict = "conflict",
+  Queue = "queue",
+}
+
+type ResponsePrepareUpdate =
+  | {
+      template: string;
+      type: ResponseType.AuthorizationRequired;
+    }
+  | {
+      template: string;
+      type: ResponseType.Conflict;
+    }
+  | {
+      queueID: number;
+      type: ResponseType.Queue;
+    };
+
+type AuthData = {
+  packageUpdateServerID: number;
+  password: string;
+  saveCredentials: boolean;
+  username: string;
+};
+
+type Payload = {
+  authData?: AuthData;
+  packages: Record<string, string>;
+};
+
+class AcpUiPackageUpdateManager implements DialogCallbackObject {
+  private readonly packages = new Map<HTMLElement, HTMLInputElement | null>();
+  private readonly submitButton: HTMLButtonElement;
+
+  constructor() {
+    this.submitButton = document.getElementById("packageUpdateSubmitButton") as HTMLButtonElement;
+
+    this.submitButton.addEventListener("click", (event) => {
+      event.preventDefault();
+
+      void this.prepareUpdate();
+    });
+
+    this.initPackages();
+  }
+
+  private initPackages(): void {
+    document.querySelectorAll(".packageUpdate").forEach((element: HTMLElement) => {
+      const checkbox = element.querySelector<HTMLInputElement>('input[type="checkbox"]');
+      checkbox?.addEventListener("change", () => this.updateSubmitButtonState());
+
+      this.packages.set(element, checkbox);
+    });
+  }
+
+  private updateSubmitButtonState(): void {
+    let disableButton = true;
+    this.packages.forEach((checkbox) => {
+      if (!checkbox || checkbox.checked) {
+        disableButton = false;
+      }
+    });
+
+    this.submitButton.disabled = disableButton;
+  }
+
+  private getSelectedPackages(): Record<string, string> {
+    const packages = {};
+    this.packages.forEach((checkbox, element) => {
+      if (checkbox === null || checkbox.checked) {
+        packages[element.dataset.package!] = element.dataset.version!;
+      }
+    });
+
+    return packages;
+  }
+
+  private async prepareUpdate(authData?: AuthData): Promise<void> {
+    const packages = this.getSelectedPackages();
+    if (Object.keys(packages).length === 0) {
+      return;
+    }
+
+    const payload: Payload = {
+      packages,
+    };
+
+    if (authData) {
+      payload.authData = authData;
+    }
+
+    let response: ResponsePrepareUpdate;
+    try {
+      this.submitButton.disabled = true;
+
+      response = (await dboAction("prepareUpdate", "wcf\\data\\package\\update\\PackageUpdateAction")
+        .payload(payload)
+        .dispatch()) as ResponsePrepareUpdate;
+    } finally {
+      this.submitButton.disabled = false;
+    }
+
+    if (response.type === ResponseType.AuthorizationRequired) {
+      this.promptCredentials(response.template);
+    } else if (response.type === ResponseType.Conflict) {
+      this.showConflict(response.template);
+    } else {
+      this.startInstallation(response.queueID);
+    }
```

Comment:
As `response.type` is an enum it would be desirable if this was a `switch()` without a `default:`. This might allow the TypeScript compiler to catch unhandled cases.